migrations/2023-09-18-141700_login-token/up.sql

```diff
@@ -0,0 +1,13 @@
+CREATE TABLE login_token (
+    id serial PRIMARY KEY,
+    token text NOT NULL UNIQUE,
+    user_id int REFERENCES local_user ON UPDATE CASCADE ON DELETE CASCADE NOT NULL,
+    published timestamptz NOT NULL DEFAULT now(),
+    ip text,
+    user_agent text
+);
+
```

Comment:
Email addresses also fall under PII, and we are storing IPs in the rate limiter. You are right that its not ideal, but practically every web service needs to store IPs to fight malicious actors.

Reply:
I'll agree that it's suboptimal. Rate limiter is a bit different because it's in-memory only, this is in the DB.

It would be good to add a comment in the struct for why it's stored (allowing listing open sessions and  invalidation by ip) and that it could be replaced by a truncated IP or derived information. Right now lemmy has more issues with spammers than with a tiny personal data storage though so it makes sense to add any tools that can help with that.

Reply:
Added